internal/file/file_test.go

```diff
@@ -40,55 +39,82 @@ func TestOpen(t *testing.T) {
 		args       args
 		want       want
 		checkFunc  func(want, *os.File) error
-		beforeFunc func(args)
-		afterFunc  func(args)
+		beforeFunc func(*testing.T, args)
+		afterFunc  func(*testing.T, args)
 	}
+
 	defaultCheckFunc := func(w want, got *os.File) error {
-		if !reflect.DeepEqual(got, w.want) {
-			return errors.Errorf("got = %v, want %v", got, w.want)
+		if w.want == nil {
+			if got != nil {
+				return errors.New("got is not nil")
+			}
+		} else {
+			if got, want := got.Name(), w.want.Name(); got != want {
+				return errors.Errorf("got name = %s, want: %s")
+			}
 		}
 		return nil
 	}
 	tests := []test{
-		// TODO test cases
-		/*
-		   {
-		       name: "test_case_1",
-		       args: args {
-		           path: "",
-		           flg: 0,
-		           perm: nil,
-		       },
-		       want: want{},
-		       checkFunc: defaultCheckFunc,
-		   },
-		*/
+		{
```

Comment:
I think you should also need to test if the directory is not exists for multiple hierarchy.